dotcom-rendering/playwright/tests/parallel-1/article.e2e.spec.ts

```diff
@@ -0,0 +1,121 @@
+import type { Response } from '@playwright/test';
+import { expect, test } from '@playwright/test';
+import { disableCMP } from '../../lib/cmp';
+import { waitForIsland } from '../../lib/islands';
+import { loadPage } from '../../lib/load-page';
+
+const responseHasJsonProperty = async (
+	response: Response,
+	url: string | RegExp,
+	expectedProperty: string,
+): Promise<boolean> => {
+	const isURL = response.request().url().match(url);
+	if (!isURL) return false;
+	const status = response.status();
+	if (status !== 200) return false;
+	const json = (await response.json()) as Record<string, string>;
+	const prop = json[expectedProperty];
+	if (!prop) return false;
+	return true;
+};
+
+test.describe('E2E Page rendering', () => {
+	test.describe('for WEB', () => {
+		test('It should load an article and make the expected network requests for dynamic content', async ({
+			context,
+			page,
+		}) => {
+			await disableCMP(context);
+
+			// rich link response promise
+			// https://api.nextgen.guardianapps.co.uk/embed/card/lifeandstyle/2013/nov/09/impostor-syndrome-oliver-burkeman.json?dcr=true
+			const richLinkResponsePromise = page.waitForResponse((response) =>
```

Comment:
`page.waitForResponse` creates a promise for the response which we await after loading the islands that trigger the requests. You can provide a predicate function to return true/false to have more control over the matched response. Here we use the predicate form to assert that the response the json message body has properties we expect.

https://playwright.dev/docs/api/class-page#page-wait-for-response

The equivalent for requests `page.waitForRequest`:
https://playwright.dev/docs/api/class-page#page-wait-for-request